inc/fields/glpiselectfield.class.php

```diff
@@ -20,6 +20,20 @@ public function prepareQuestionInputForSave($input) {
       return $input;
    }
 
+   public function isValid($value) {
+      // If the field is required it can't be empty (0 is a valid value for entity)
+      if ($this->isRequired() && empty($value) && ($value == '0' && $this->fields['values'] != Entity::class)) {
```

Comment:
do not use `empty`, it a bad function in php, you should list all case where a select is not selected (0 like after, maybe '---', etc)

Reply:
for information, `(empty($var))` is like doing `($var == false)`

Reply:
This is a technical debt. I overloaded the method from the parent class. I prefer not to change its way too much to avoid regressions.

I'll probably drop use of empty() when I'll refactor all the fields classes. For now I still focus on bugfixes and finishing pending PRs

Reply:
ok fine, review ok in this case